drivers/i2c/i2c_ll_stm32.c

```diff
@@ -93,6 +93,9 @@ static int i2c_stm32_init(struct device *dev)
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	u32_t bitrate_cfg;
 	int ret;
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+	struct i2c_stm32_data *data = DEV_DATA(dev);
+#endif
```

Comment:
@tbursztyka I agree with you that it will make the code more readable. The problem is that it will fix compilation for interrupt mode but it will break compilation for polling mode. The device_sync_sem and irq_config_func are added to structs  i2c_stm32_data and i2c_stm32_config only for interrupt mode.